src/ffi.rs

```diff
@@ -627,4 +820,74 @@ mod tests {
             assert!(errbuf_len < ERR_BUF_MAX_LEN);
         }
     }
+
+    #[cfg(feature = "expr_validation")]
+    #[test]
+    fn test_expression_validate() {
```

Comment:
I think we'd better add test cases for `ATC_ROUTER_EXPRESSION_VALIDATE_FAILED` and `ATC_ROUTER_EXPRESSION_VALIDATE_BUF_TOO_SMALL`.

Reply:
fixed.